salt/cloud/clouds/digital_ocean.py

```diff
@@ -720,20 +772,18 @@ def destroy(name, call=None):
     return node
 
 
-def create_dns_record(hostname, ip_address):
+def post_dns_record(dns_domain, name, record_type, record_data):
```

Comment:
Thanks for the onboarding guidance! I'll add the deprecation warnings.

I think this needs to change. A) create_dns_record() as used should be deprecated. I'll try to justify that first. B) it's not clear that this function should be an API function, and I suggest it probably should not be. I'll try to justify that second.

There are lots of DNS records that have nothing to do with hostnames: SOA, NS, TXT, SRV for example. It would be wrongheaded to design the API to prevent their use. There are at least two types of DNS records that need to be unambiguously distinct: A and AAAA.

In the case of DigitalOcean, their REST interface is a POST HTTP method. We're wrapping that, and calling it "create" is cargo-cultish, and I'm changing the arguments, so presumably we *could* have both. IMHO, at the very least create_dns_record() should be changed create_dns_A_record() if you're into Knuth-y literate programming.

Also 'create_dns_record' is a nice declarative to use in the context of a VM config. There is a convention in the cloud drivers to represent those as identical variable identifiers. That would preclude using the configuration item in the same variable scope as the function that implements it: ouch. As far as API compatibility: I don't think anything outside of the DigitalOcean driver code calls this. AFAICT the upstream exposure is through salt-cloud config options.